src/Internal/Plutus/Types/Value.purs

```diff
@@ -82,6 +87,18 @@ instance EncodeAeson Value where
   encodeAeson' (Value mph) = encodeAeson' $ encodeAeson
     { "getValue": encodeAeson mph }
 
+arbitrarySingletonValue :: Gen Value
+arbitrarySingletonValue = do
+  currencySymbol <- unMaybeGen $ mkCurrencySymbol <$> arbitrary
+  tokenName <- unMaybeGen $ mkTokenName <$> arbitrary
+  num <- BigInt.fromInt <$> arbitrary
+  pure $ singleton currencySymbol tokenName num
+
+instance Arbitrary Value where
```

Comment:
Did you look at plutus or cardano or someone else to write this instance? (I'm curious).

Also, I don't think we need wholly arbitrary names for the `CurrencySymbol`, you can use a fixed set of names with the `Test.QuickCheck.Gen.frequency` function instead. This would allow you to remove the need for `unMaybeGen` and with it the new module `Internal.Quickcheck` .

Reply:
What is wrong with having `Internal.Quickcheck` module? It is not a multilayer abstraction it is a simple utility module, which is used only in tests. Maybe it bloating bundle size, I do not know, maybe than move it to tests dir or something.

Reply:
I do not see `Arbitrary` instance in API docs. Maybe I am looking not in the right place?

https://playground.plutus.iohkdev.io/doc/haddock/plutus-ledger-api/html/Plutus-V1-Ledger-Value.html

Reply:
I don't know, once I tried to provide `Arbitrary` instances for them and other types, there was a discussion a bout how to do it properly, but at the end I didn't proceed to do it. To me this instance looks good.